lib/docker2aci.go

```diff
@@ -528,17 +534,16 @@ func writeACI(layer io.ReadSeeker, manifest schema.ImageManifest, output string)
 	trw := tar.NewWriter(gw)
 	defer trw.Close()
 
-	if err := addMinimalACIStructure(trw, manifest); err != nil {
-		return fmt.Errorf("error writing rootfs entry: %v", err)
-	}
-
+	var whiteouts []string
 	convWalker := func(t *tarball.TarFile) error {
 		name := t.Name()
 		if name == "./" {
 			return nil
 		}
 		t.Header.Name = path.Join("rootfs", name)
+		absolutePath := strings.TrimPrefix(t.Header.Name, "rootfs")
 		if strings.Contains(t.Header.Name, "/.wh.") {
+			whiteouts = append(whiteouts, strings.Replace(absolutePath, ".wh.", "", -1))
```

Comment:
I don't know the specific legalities of filenames here, but wouldn't `1` be safer than `-1`? (e.g. `.wh.myfile.wh.ich.has.a.silly.name`)